doc/SAI-Proposal-Packet-Trimming.md

```diff
@@ -0,0 +1,141 @@
+# Switch Abstraction Interface Change Proposal for Packet Trimming
+
+Title       | Packet Trimming
+------------|----------------
+Authors     | Nvidia
+Status      | In review
+Type        | Standards track
+Created     | 8/28/2024
+SAI-Version | 1.14
+----------
+
+## Overview
+When the lossy queue exceeds a buffer threshold, it drops packets without any notification to the destination host.
+
+When a packet is lost, it can be recovered through fast retransmission (e.g., Go-Back-N in RoCE) or by using timeouts. Retransmission triggered by timeouts typically incurs significant latency. Packet trimming aims to facilitate rapid packet loss notification and, consequently, eliminate slow timeout-based retransmissions.
+
+To help the host recover data more quickly and accurately, we introduce a packet trimming feature, that upon a failed packet admission to a shared buffer,
+will trim a packet to a configured size, and try sending it on a different queue to deliver a packet drop notification to an end host.
```

Comment:
What will the trimmed packet's headers look like? Will it be the same as the original packet with the exception of the length and DSCP information?

Reply:
> What will the trimmed packet's headers look like? Will it be the same as the original packet with the exception of the length and DSCP information?

Yes, TRIM packets are original packets, trimmed to the specified length and IP length updated